src/transitions/Fade.js

```diff
@@ -1,53 +1,60 @@
 // @flow weak
 
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
+import React, { Element, PureComponent } from 'react';
 import Transition from '../internal/Transition';
 import { duration } from '../styles/transitions';
 import customPropTypes from '../utils/customPropTypes';
 
-export default class Fade extends Component {
-  static propTypes = {
-    children: PropTypes.node,
-    /**
-     * If `true`, the component will transition in.
-     */
-    in: PropTypes.bool,
-    /**
-     * Duration of the animation when the element is entering.
-     */
-    enterTransitionDuration: PropTypes.number, // eslint-disable-line react/sort-prop-types
-    /**
-     * Duration of the animation when the element is exiting.
-     */
-    leaveTransitionDuration: PropTypes.number,
-    /**
-     * Callback fired before the component enters.
-     */
-    onEnter: PropTypes.func,
-    /**
-     * Callback fired when the component is entering.
-     */
-    onEntering: PropTypes.func,
-    /**
-     * Callback fired when the component has entered.
-     */
-    onEntered: PropTypes.func, // eslint-disable-line react/sort-prop-types
-    /**
-     * Callback fired before the component exits.
-     */
-    onExit: PropTypes.func,
-    /**
-     * Callback fired when the component is exiting.
-     */
-    onExiting: PropTypes.func,
-    /**
-     * Callback fired when the component has exited.
-     */
-    onExited: PropTypes.func, // eslint-disable-line react/sort-prop-types
-  };
+type DefaultProps = {
+  in: boolean,
+  enterTransitionDuration: number,
+  leaveTransitionDuration: number,
+}
+
+type Props = DefaultProps & {
+  children?: Element<*>,
+  /**
+   * If `true`, the component will transition in.
+   */
+  in?: boolean,
+  /**
+   * Duration of the animation when the element is entering.
+   */
+  enterTransitionDuration?: number, // eslint-disable-line react/sort-prop-types
+  /**
+   * Duration of the animation when the element is exiting.
+   */
+  leaveTransitionDuration?: number,
+  /**
+   * Callback fired before the component enters.
+   */
+  onEnter?: Function,
+  /**
+   * Callback fired when the component is entering.
+   */
+  onEntering?: Function,
+  /**
+   * Callback fired when the component has entered.
+   */
+  onEntered?: Function, // eslint-disable-line react/sort-prop-types
+  /**
+   * Callback fired before the component exits.
+   */
+  onExit?: Function,
+  /**
+   * Callback fired when the component is exiting.
+   */
+  onExiting?: Function,
+  /**
+   * Callback fired when the component has exited.
+   */
+  onExited?: Function, // eslint-disable-line react/sort-prop-types
+};
+
+export default class Fade extends PureComponent<DefaultProps, Props, void> {
```

Comment:
This was a `Component` as was the other you mentioned, so you could say this is an improvement over there former.  This time there is quite more in the body of the class where I much prefer class closure from a source perspective.  Let's be a bit incremental about the codebase, I unfortunately don't always have time to take the former code and make every PR exactly inline. Each PR author needs to be able to improve the code an increment without taking on a burden of everything that might be wrong with a file.